File: docs/vs-2015/snippets/csharp/VS_Snippets_ProTools/consentdialog/cs/form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace ConsentDialog
{
    public partial class Form1 : Form
    {
        //<snippet6>
        public Form1()
        {
            InitializeComponent();
            ProceedButton.Enabled = false;
        }
        //</snippet6>

        //<snippet3>
        public bool accepted = false;
        //</snippet3>

        //<snippet1>
        private void checkBox1_CheckedChanged(object sender, EventArgs e)
        {
            ProceedButton.Enabled = !ProceedButton.Enabled;
        }
        //</snippet1>

        //<snippet2>
        private void ProceedButton_Click(object sender, EventArgs e)
        {
            if (ProceedButton.Enabled)
            {
                accepted = true;
                this.Close();
            }
        }
        //</snippet2>

        //<snippet4>
        private void CancelButton_Click(object sender, EventArgs e)
        {
            this.Close();
        }
        //</snippet4>



    }
}

File: docs/vs-2015/snippets/visualbasic/VS_Snippets_ProTools/consentdialog/vb/form1.vb
Public Class Form1

    '<snippet6>
    Public Sub New()
        InitializeComponent()
        ProceedButton.Enabled = False
    End Sub
    '</snippet6>

    '<snippet3>
    Public accepted As Boolean = False
    '</snippet3>

    '<snippet1>
    Private Sub CheckBox1_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles CheckBox1.CheckedChanged
        ProceedButton.Enabled = Not ProceedButton.Enabled
    End Sub
    '</snippet1>

    '<snippet2>
    Private Sub ProceedButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ProceedButton.Click
        If ProceedButton.Enabled Then
            accepted = True
            Me.Close()
        End If
    End Sub
    '</snippet2>

    '<snippet4>
    Private Sub CancelButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles CancelButton.Click
        Me.Close()
    End Sub
    '</snippet4>
End Class
